packages/react/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,174 @@
+import { forwardRef } from 'react';
+import * as React from 'react';
+import cn from 'classnames';
+import { ChevronLeftIcon, ChevronRightIcon } from '@navikt/aksel-icons';
+
+import { Button } from '../Button';
+
+import classes from './Pagination.module.css';
+
+export interface PaginationProps {
+  /** Sets the text label for the next page button
+   * @default ''
+   */
+  nextLabel: string;
+  /** Sets the text label for the previous page button
+   * @default ''
+   */
+  previousLabel: string;
+  /** Sets the size of the component
+   * @default 'medium'
+   */
+  size?: 'small' | 'medium';
+  /** Sets how compact the component will be. If true, only 5 steps will show.
+   * @default false
+   */
+  compact?: boolean;
+  /** Hides the component's previous and next button labels
+   * @default false
+   */
+  hideLabels?: boolean;
+  /** Sets the current page */
+  currentPage: number;
+  /** Total number of pages */
+  totalPages: number;
+  /**
+   * Function to be called when the selected page changes.
+   */
+  onChange: (currentPage: number) => void;
+}
+
+// eslint-disable-next-line react/display-name
+export const Pagination = forwardRef<HTMLElement, PaginationProps>(
+  (
+    {
+      nextLabel = '',
+      previousLabel = '',
+      size = 'small',
+      compact = false,
+      hideLabels = false,
+      currentPage = 1,
+      totalPages,
+      onChange,
+    }: PaginationProps,
+    ref,
+  ) => {
+    const getSteps = () => {
```

Comment:
Suggest moving this function outside and making it pure. Not only will it be easier to test, its a hefty function to declare on every render call by react.

Reply:
What about memoizing it? I think the component test already tests a lot of the function

Reply:
Hmm, maybe both? You can also add types to it without it getting to cluttery. Setting the return type to `('ellipsis'| number)[]` will also give you type safety and make the ternary check for step in your markup cleaner :)

Reply:
I'll just move it outside as memo only makes sense when the function is inside the component